__tests__/notifications.ts

```diff
@@ -750,3 +819,238 @@ describe('mutation clearNotificationPreference', () => {
+      .findOneBy(params);
+
+    expect(notificationPreference).not.toBeNull();
+    expect(notificationPreference!.type).toEqual(
+      notificationPreferenceMap[params.notificationType],
+    );
+    expect(notificationPreference!.status).toEqual(
+      NotificationPreferenceStatus.Subscribed,
+    );
+  });
+
+  it('should set notification preference to subscribed and fetch the reference id if it is article new comment or squad new comment', async () => {
+    loggedUser = '1';
+
+    await prepareNotificationPreferences({
+      status: NotificationPreferenceStatus.Subscribed,
+    });
+    const comment = {
+      id: 'c1',
+      postId: postsFixture[0].id,
+      content: '',
+      userId: '1',
+    };
+    await con.getRepository(Comment).save(comment);
+
+    const params = {
+      userId: loggedUser,
+      referenceId: 'c1',
+      notificationType: NotificationType.ArticleNewComment,
+    };
+
+    const preference = await con
+      .getRepository(NotificationPreference)
+      .findOneBy(params);
+
+    expect(preference).toBeFalsy();
+
+    await client.mutate(MUTATION, {
+      variables: {
+        referenceId: params.referenceId,
+        type: params.notificationType,
+      },
+    });
+
+    const notificationPreference = await con
+      .getRepository(NotificationPreference)
+      .findOneBy({
+        notificationType: params.notificationType,
+        referenceId: comment.postId,
+        status: NotificationPreferenceStatus.Subscribed,
+      });
+
+    expect(notificationPreference).toBeTruthy();
+    expect(notificationPreference!.type).toEqual(
+      notificationPreferenceMap[params.notificationType],
+    );
+    expect(notificationPreference!.status).toEqual(
+      NotificationPreferenceStatus.Subscribed,
+    );
+  });
+
+  it('should ignore when preference is already subscribed', async () => {
+    loggedUser = '1';
+
+    await prepareNotificationPreferences({
+      status: NotificationPreferenceStatus.Subscribed,
+    });
+
+    const params = {
+      userId: loggedUser,
+      referenceId: postsFixture[2].id,
+      notificationType: NotificationType.ArticleNewComment,
+    };
+
+    await client.mutate(MUTATION, {
+      variables: {
+        referenceId: params.referenceId,
+        type: params.notificationType,
+      },
+    });
+
+    const preference = await con
+      .getRepository(NotificationPreference)
+      .findOneBy(params);
+
+    expect(preference).toBeTruthy();
+
+    await client.mutate(MUTATION, {
+      variables: {
+        referenceId: params.referenceId,
+        type: params.notificationType,
+      },
+    });
+
+    const notificationPreference = await con
+      .getRepository(NotificationPreference)
+      .findOneBy(params);
+
+    expect(notificationPreference).toBeTruthy();
+    expect(notificationPreference!.status).toEqual(
+      NotificationPreferenceStatus.Subscribed,
+    );
+  });
+
+  it('should update status when different preference is set', async () => {
```

Comment:
This is the test that confirms https://github.com/dailydotdev/daily-api/pull/1561/files#r1411843132